### DISCUSSION TITLE
eslint-config-prettier

### DISCUSSION THREAD

**Discussion by alexander-densley** [t3-oss/create-t3-turbo#982] [Q&A]
Is there a reason that the eslint-config-prettier plugin for eslint is not used? Is it redundant or was it a choice not to use this plugin.

**Comment by noxify** [comment]
Was removed in february via https://github.com/t3-oss/create-t3-turbo/pull/875

**Reply by alexander-densley** [reply]
does this make it so when you save a file it doens't format things?

**Reply by alexander-densley** [reply]
cause when I save there is no formating going on

**Reply by noxify** [reply]
Haven't updated my local repo with the latest changes ( & upgrading to eslint ), but with eslint 8.57.0 it works

**Comment by juliusmarminge** [comment]
https://www.joshuakgoldberg.com/blog/you-probably-dont-need-eslint-config-prettier-or-eslint-plugin-prettier/

**Reply by alexander-densley** [reply]
would you happen to know why formatting and linting is not occurring? Ive brought over the tooling workspaes exactly as the start repo has, and have copied teh .vscode settings json and have updated all the package.jsons to import your toolings and updated all the config files, is there something im missing to port that over?

**Reply by alexander-densley** [reply]
Also, I'm sorry if that isn't enough to go on, happy to drop my repo link or anything else. been working on just copying the tooling packages into my turbo repo for over 5 hours and im banging my head agaisnt the desk at this point. everything is the same and it works in your repo that I git cloned but not in mine

**Reply by noxify** [reply]
* Does it work with a fresh installation?
* Also, what errors do you get in the "Output" tab inside VSC?
* A Repro would be good :)

**Reply by alexander-densley** [reply]
when i git clone the template repo it works, but when I try to implement the tooling and configs in my already existing repo, it doesn't work. there are no errors shown, it's like there is no lint at all even being attempted.

**Reply by alexander-densley** [reply]
wait, just found this:
["ERROR" - 2:11:30 PM] Error handling text editor change
["ERROR" - 2:11:30 PM] Error resolve node module '@ianvs/prettier-plugin-sort-imports'
Error: Error resolve node module '@ianvs/prettier-plugin-sort-imports'

**Reply by noxify** [reply]
> Update: you was faster than me :) 

Just to make sure I understood it correctly:

After choosing "Prettier" in the dropdown, you don't see any errors - And after saving you also don't see any errors?
Here an example what I see in my prettier output:
<img width="1057" alt="image" src="https://github.com/t3-oss/create-t3-turbo/assets/521777/75be2847-fc88-48f2-87ad-2df88fc17011">

Same for eslint:
<img width="1079" alt="image" src="https://github.com/t3-oss/create-t3-turbo/assets/521777/db85319a-33b0-44f1-a827-8b96c2e27133">

**Reply by noxify** [reply]
I think it would be good if you could provide a repo :)

**Reply by alexander-densley** [reply]
[branch link removed]
here is the branch

also, just discovered that everything works fine in the prettier package, but nowhere else

**Reply by alexander-densley** [reply]
looks like i get errors for all of the prettier extensions like:
Error: Error resolve node module 'prettier-plugin-tailwindcss'

**Reply by noxify** [reply]
Fun fact: The prettier seems to work inside the "tooling" directory. 
But if you save a file inside a different directory ( e.g. `packages/db/supabase-client.ts` ), you will get the error.

**Reply by alexander-densley** [reply]
wait, does it work in all tooling folders? eslint and tailwind and all that?

**Reply by alexander-densley** [reply]
cause it doesn't work for me in those areas. only in the actual prettier folder. also its not set up in db or api. it should be set up in the apps/web, but it isn't working there either


also, thank you os much for the help, 🙏

**Reply by noxify** [reply]
pushed my changes to: https://github.com/noxify/subtract/tree/ad-ENG-41-tooling-package

If I change a file inside tooling/eslint - The formatting is working.

![Kapture 2024-04-10 at 23 01 23](https://github.com/t3-oss/create-t3-turbo/assets/521777/25e940c8-984a-4c1e-aea4-8055a1d2a388)

**Reply by noxify** [reply]
For the eslint package, I get the following error in the output:

```
Uncaught exception received.
Error: Could not find config file.
    at locateConfigFileToUse (/.../subtract/node_modules/.pnpm/eslint@9.0.0/node_modules/eslint/lib/eslint/eslint.js:349:21)
    at async calculateConfigArray (/.../subtract/node_modules/.pnpm/eslint@9.0.0/node_modules/eslint/lib/eslint/eslint.js:384:49)
    at async ESLint.calculateConfigForFile (/.../subtract/node_modules/.pnpm/eslint@9.0.0/node_modules/eslint/lib/eslint/eslint.js:1173:25)
    at async ESLint.isPathIgnored (/.../subtract/node_modules/.pnpm/eslint@9.0.0/node_modules/eslint/lib/eslint/eslint.js:1197:24)
    at async /Users/marcusreinhardt/.vscode/extensions/dbaeumer.vscode-eslint-2.4.4/server/out/eslintServer.js:1:23897
    at async Object.E [as withClass] (/Users/marcusreinhardt/.vscode/extensions/dbaeumer.vscode-eslint-2.4.4/server/out/eslintServer.js:1:19114)
    at async T.then.g.validate (/Users/marcusreinhardt/.vscode/extensions/dbaeumer.vscode-eslint-2.4.4/server/out/eslintServer.js:1:23868)
    at async /Users/marcusreinhardt/.vscode/extensions/dbaeumer.vscode-eslint-2.4.4/server/out/eslintServer.js:1:220163

```

**Reply by noxify** [reply]
but tbh - i have no clue why it's not working - maybe @juliusmarminge has an idea what could be wrong.

**Reply by alexander-densley** [reply]
thanks man, I appreciate it! maybe @juliusmarminge will have some thoughts

**Reply by RedBeardEth** [reply]
You may need to update your code editors eslint extension to use the pre-release version (v3.0.5) for v9 support
![image](https://github.com/t3-oss/create-t3-turbo/assets/90423049/6ea67e78-e329-41a9-bee9-d7f3a43fbfa3)

**Reply by alexander-densley** [reply]
@noxify , hey, super random, would you mind deleting your local copy and fork of the repo? I only made my repo public for just yesterday, but now it's private again, would you be able to do that? thanks

**Reply by stunaz** [reply]
Same issue here, switched to pre-release. The cli seems to work. But the ide (vscode) does not highlight anything. (mainly talking about eslint tho)

**Reply by noxify** [reply]
> @noxify , hey, super random, would you mind deleting your local copy and fork of the repo? I only made my repo public for just yesterday, but now it's private again, would you be able to do that? thanks

Will do that when I'm back in the office 

Edit: Done

**Reply by alexander-densley** [reply]
i got mine working today, not really sure what i did wrong or right, but some things to do for sure, switch to pre release version, copy exactly what is in the repo, it can help to delete and reinstall all your packages as well as restart vscode. good luck to anyone else that runs into these weird bugs
